flasc/energy_ratio/energy_ratio.py

```diff
@@ -504,6 +509,8 @@ def _get_energy_ratios_all_wd_bins_bootstrapping(
     N=1,
     percentiles=[5.0, 95.0],
     return_detailed_output=False,
+    block_bootstrapping = False, 
+    num_blocks = 10
```

Comment:
I wonder if somehow we can make this more uniform. Essentially, regular bootstrapping is block bootstrapping with num_blocks==num_samples, correct? Maybe that can simplify things like that and that comparison can also work as a unit test.

Then, we can collapse the two options `block_bootstrapping` and `num_blocks` just to `num_bootstrapping_blocks`, with `-1` being the default option meaning `num_blocks=num_samples`? Or maybe that would all become too ambiguous. What do you think?

Reply:
I like it in principle, my only doubt is if the block-bootstrapped version of normal bootstrapping with num_blocks=num_samples is slower than the present implementation?  I should test, if that's ok, I think it makes sense.  What do you think @misi9170 ?

Reply:
I agree, would be good to make sure things don't slow down!

Reply:
hi @Bartdoekemeijer  I tried this by putting in this code block around line 670 in energy_ratio.py:

```
        if not block_bootstrapping:
            num_blocks = df_binned.shape[0]
            block_bootstrapping = True
```
Unfortunately it was about 10x slower, I think maybe because the other call functions through built-in pandas (numpy) methods it gets a big speed up.   So maybe we keep the logic as is? 

 Still though, I think we could have the function inputs act as you like, instead of the seperate boolean, just haave num_blocks default to -1, meaning no blocks, does that work for you?

Reply:
Yes, that works! Thanks for checking, and I agree that a 10x slowdown is not worth it.

Reply:
I'd say feel free to merge if this does not affect current code functionality. Given the slowdown, it's best to keep the bootstrapping and block bootstrapping separate.